src/main/java/frc/robot/subsystems/climb/ClimbIOSim.kt

```diff
@@ -0,0 +1,54 @@
+package frc.robot.subsystems.climb
+
+import edu.wpi.first.math.system.plant.DCMotor
+import edu.wpi.first.math.util.Units
+import edu.wpi.first.units.Units.*
+import edu.wpi.first.units.measure.Distance
+import edu.wpi.first.units.measure.MomentOfInertia
+import edu.wpi.first.units.measure.MutVoltage
+import edu.wpi.first.units.measure.Voltage
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim
+import lib.math.units.into
+import kotlin.math.PI
+
+class ClimbIOSim(
+    motor: DCMotor,
+    gearing: Double,
+    moi: MomentOfInertia,
+    armLength: Distance
+): ClimbIO {
+
+    private val armSim: SingleJointedArmSim = SingleJointedArmSim(
+        motor,
+        gearing,
+        moi into KilogramSquareMeters,
+        armLength into Meters,
+        Units.degreesToRadians(-45.0),
+        Units.degreesToRadians(180.0 + 45.0),
+        true,
+        PI / 2
+    )
+
+    private val appliedVoltage: MutVoltage = Volts.zero().mutableCopy()
+
+    override fun updateInputs(inputs: ClimbIO.ClimbArmInputs) {
+
+        armSim.update(0.02)
```

Comment:
see above (I know that this is the update rate but it shouldn't just be a magic number)